### PR TITLE
Setup mongoDB, requests and routes

### PR DESCRIPTION
**Pull request by ncbecker**
## Related Issue

#35

### CODE REVIEW

File: client/src/components/EpisodeSearch.js [hunk 1]
```diff
@@ -65,10 +65,10 @@ export const EpisodeSearch = ({ value, onChange, suggestions, onClick }) => {
       />
       {suggestions && (
         <SuggestionsContainer>
-          {suggestions?.map((episode) => (
+          {suggestions.map((episode) => (
             <SearchSuggestion key={episode.id} onClick={onClick}>
               <span>{episode.name}</span>
-              <img src={episode.images[2].url} alt={episode.name} />
+              <img src={episode.images[2]?.url} alt={episode.name} />
```
Comment (boldCM) [inline client/src/components/EpisodeSearch.js:80]: just curious: how come you want this specific index?

File: client/src/utils/api.js [hunk 2]
```diff
@@ -28,6 +32,32 @@ export async function searchEpisode(q) {
...
+  const response = await fetch("/api/db/users");
+  const allUsers = await response.json();
+  return allUsers;
+}
+
+export async function getAllEpisodes(id) {
+  const response = await fetch("/api/db/episodes");
+  const allEpisodes = await response.json();
+  return allEpisodes;
+}
```
Comment (boldCM) [inline client/src/utils/api.js]: Good Job on naming!👍

**Review by boldCM — APPROVED** [review]
lgtm 🎉

File: client/src/components/EpisodeSearch.js (hunk 1 above)
Comment (phlgr) [inline client/src/components/EpisodeSearch.js:80]: Doesn't have every episode a url?

File: client/src/pages/LogInPage.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ function LogInPage() {
         <Logo />
       </LogoWrapper>
       <ButtonWrapper>
-        <a href="/oauth/spotify/authorize">
+        <a href="http://localhost:3001/oauth/spotify/authorize">
```
Comment (phlgr) [inline client/src/pages/LogInPage.js:39]: `setupProxy.js` would be useful

File: client/src/utils/api.js (hunk 2 above)
Comment (phlgr) [inline client/src/utils/api.js]: Nice and clean

File: lib/database.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+let db;
+
+async function connectDb(url, dbName) {
+  client = await MongoClient.connect(url, { useUnifiedTopology: true });
+  db = client.db(dbName);
+}
+
+function disconnectDb() {
+  client.close();
+}
```
Comment (phlgr) [inline lib/database.js:13]: Do you ever use this?

File: server.js [hunk 5]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+      .status(500)
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.get("/api/db/user/:id", async (request, response) => {
+  const { id } = request.params;
+  try {
+    const singleUser = await getSingleUser(id);
+    response.send(singleUser);
```
Comment (phlgr) [inline server.js]: Also send the status here
```suggestion
    response.status(200).send(singleUser);
```

File: server.js [hunk 6]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.patch("/api/db/user/:id", async (request, response) => {
+  const { id } = request.params;
+  const { name } = request.body;
+  try {
+    const updatedUser = await updateUserDisplayName(id, name);
+    response.send(updatedUser);
```
Comment (phlgr) [inline server.js]: Same here 
```suggestion
    response.status(200).send(updatedUser);
```

File: server.js [hunk 7]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+    response
+      .status(500)
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.get("/api/db/users", async (request, response) => {
+  try {
+    const allUsers = await getAllUsers();
+    response.send(allUsers);
```
Comment (phlgr) [inline server.js]:
```suggestion
    response.status(200).send(allUsers);
```

File: server.js [hunk 8]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+      .status(500)
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.get("/api/db/episode/:id", async (request, response) => {
+  const { id } = request.params;
+  try {
+    const singleEpisode = await getSingleEpisode(id);
+    response.send(singleEpisode);
```
Comment (phlgr) [inline server.js]:
```suggestion
    response.statu(200).send(singleEpisode);
```

File: server.js [hunk 9]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+      .status(500)
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.delete("/api/db/episode/:id", async (request, response) => {
+  const { id } = request.params;
+  try {
+    const deletedEpisode = await deleteEpisode(id);
+    response.send(deletedEpisode);
```
Comment (phlgr) [inline server.js]:
```suggestion
    response.status(200).send(deletedEpisode);
```

File: server.js [hunk 10]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+    response
+      .status(500)
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.get("/api/db/episodes", async (request, response) => {
+  try {
+    const allEpisodes = await getAllEpisodes();
+    response.send(allEpisodes);
```
Comment (phlgr) [inline server.js]:
```suggestion
    response.status(200).send(allEpisodes);
```

File: server.js [hunk 11]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+      .status(500)
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.get("/api/db/episodes/:id", async (request, response) => {
+  const { id } = request.params;
+  try {
+    const allLikedEpisodes = await getAllLikedEpisodes(id);
+    response.send(allLikedEpisodes);
```
Comment (phlgr) [inline server.js]:
```suggestion
    response.status(200).send(allLikedEpisodes);
```

File: server.js [hunk 12]
```diff
@@ -163,6 +180,169 @@ app.get("/api/episode/:id", async (request, response) => {
...
+    response
+      .status(500)
+      .send("An unexpected error occured. Please try again later!");
+  }
+});
+
+app.delete("/api/db/episodes", async (request, response) => {
+  try {
+    const deletedEpisodes = await deleteManyEpisodes();
+    response.send(deletedEpisodes);
```
Comment (phlgr) [inline server.js]:
```suggestion
    response.status(200).send(deletedEpisodes);
```

**Review by phlgr — CHANGES_REQUESTED** [review]

**Pull request merged**